sqlitecluster/SQLiteNode.cpp

```diff
@@ -2744,12 +2744,30 @@ SQLitePeer* SQLiteNode::_getPeerByID(uint64_t id) const {
     }
 }
 
-SQLitePeer* SQLiteNode::_getPeerByName(const string& name) const {
+bool comparePeers(SQLitePeer peer1, SQLitePeer peer2) {
+    return peer1.name < peer2.name;
+}
+
+int SQLiteNode::binarySearchPeers(const string& name) {
+    auto it = std::lower_bound(_peerList.begin(), _peerList.end(), name);
+    if (it == _peerList.end() || it.operator*()->name != name) {
+        return -1;
+    } else {
+        std::size_t index = std::distance(_peerList.begin(), it);
+        return index;
+    }
```

Comment:
You are early-returning in the `if` so you don't need an `else` here, you can break it out. Also I don't think there's a need to save `index` to a variable, since you are not doing anything with it - you can just return the result of `distance()`